plugins/ruby/plugin.zsh

```diff
@@ -0,0 +1,65 @@
+# Color definitions
+GEOMETRY_COLOR_RUBY_RVM_VERSION=${GEOMETRY_COLOR_PROMPT:-white}
+
+GEOMETRY_RUBY_RVM_SHOW_GEMSET=${GEOMETRY_RUBY_RVM_SHOW_GEMSET:-true}
+
+# Symbol definitions
+GEOMETRY_SYMBOL_RUBY_RVM_VERSION=${GEOMETRY_SYMBOL_RUBY_RVM_VERSION:-"◆"}
+GEOMETRY_RUBY_RVM_VERSION=$(prompt_geometry_colorize $GEOMETRY_COLOR_RUBY_RVM_VERSION $GEOMETRY_SYMBOL_RUBY_RVM_VERSION) 
+
+get_full_ruby_version() {
+  if (( $+commands[ruby] )) && [[ -z $GEOMETRY_RUBY_VERSION_FULL ]]; then
+      GEOMETRY_RUBY_VERSION_FULL="$(ruby -v)"
+  fi
+}
+
+get_full_rvm_version() {
+  if (( $+commands[rvm] )) && [[ -z $GEOMETRY_RVM_VERSION_FULL ]]; then
+      GEOMETRY_RVM_VERSION_FULL="$(rvm -v)"
+  fi
+}
+
+geometry_prompt_ruby_setup() {
+  GEOMETRY_RUBY_VERSION_FULL="$(ruby -v)"
```

Comment:
Can't we use get_full_ruby_version function for this? To avoid code duplication.